core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -474,6 +475,12 @@ final class ShuffleBlockFetcherIterator(
               // TODO: manage the memory used here, and spill it into disk in case of OOM.
               Utils.copyStream(input, out, closeStreams = true)
               input = out.toChunkedByteBuffer.toInputStream(dispose = true)
+            } else {
+              val readAheadEnabled = SparkEnv.get != null &&
```

Comment:
I think we'd better not use `SparkEnv` here, though it can not be null in actual execution.
We can put this config a constructor args of the `ShuffleBlockFetcherIterator` just like the `detectCorrupt` config.

Reply:
Actually,  `detectCorrupt ` is got by ` SparkEnv.get.conf.get(config.SHUFFLE_DETECT_CORRUPT)`,.
Although this is consistent with before,  it brings more changes.

Reply:
@10110346 
Yes, I know it's got through `SparkEnv`, but we should put the call the same place to be consistent, or the code is messed up, and I the changes only involves a few of classes, and it's more natural. If you hold on your opinion, may be we should see other people's suggestion.